ballerina-tests/tests/http_specific_media_type_prefix_test.bal

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+import ballerina/http;
+
+http:Client serviceSpecificClientEP = check new("http://localhost:" + serviceMediaTypeSubtypePrefixPort.toString());
+listener http:Listener serviceSpecificListener = new(serviceMediaTypeSubtypePrefixPort);
+
+ @http:ServiceConfig {
+         mediaTypeSubtypePrefix : "testServicePrefix1"
+ }
+service /service1 on serviceSpecificListener {
+
+    resource function default test1(http:Request req) returns string {
+        return "test1";
+    }
+
+    resource function default test2(http:Request req) returns @http:Payload{mediaType : "type2/subtype2"} string {
+            return "test2";
+    }
+
+    resource function default test3(http:Request req) returns http:Response {
+        http:Response res = new;
+        res.setPayload("test3");
+        checkpanic res.setContentType("type3/subtype3");
+        return res;
+    }
+
+    resource function default test4(http:Request req) returns http:Response {
+        http:Response res = new;
+        res.setPayload("test4", "type4/subtype4");
+        return res;
+    }
+
+    resource function default test5(http:Request req) returns http:Response {
+        http:Response res = new;
+        res.setPayload("test5");
+        res.setHeader("content-type", "type5/subtype5");
+        return res;
+    }
```

Comment:
Let's add a test case for a resource which send response using caller respond method also